pkg/kubectl/cmd/drain.go

```diff
@@ -0,0 +1,313 @@
+// any unmanaged pods and the user didn't pass --force, we return that list in
+// an error.
+func (o *DrainOptions) getPodsForDeletion() ([]api.Pod, error) {
+	pods := []api.Pod{}
+	podList, err := o.client.Pods(api.NamespaceAll).List(api.ListOptions{FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": o.nodeInfo.Name})})
+	if err != nil {
+		return pods, err
+	}
+	unreplicatedPodNames := []string{}
+
+	for _, pod := range podList.Items {
+		_, found := pod.ObjectMeta.Annotations[types.ConfigMirrorAnnotationKey]
+		if found {
+			// Skip mirror pod
+			continue
+		}
+		replicated := false
+
+		creatorRef, found := pod.ObjectMeta.Annotations[controller.CreatedByAnnotation]
+		if found {
+			// Now verify that the specified creator actually exists.
+			var sr api.SerializedReference
+			err := api.Scheme.DecodeInto([]byte(creatorRef), &sr)
+			if err != nil {
+				return pods, err
+			}
+			if sr.Reference.Kind == "ReplicationController" {
+				rc, err := o.client.ReplicationControllers(sr.Reference.Namespace).Get(sr.Reference.Name)
+				// Assume the only reason for an error is because the RC is
+				// gone/missing, not for any other cause.  TODO(mml): something more
+				// sophisticated than this
+				if err == nil && rc != nil {
+					replicated = true
+				}
+			} else if sr.Reference.Kind == "DaemonSet" {
+				ds, err := o.client.DaemonSets(sr.Reference.Namespace).Get(sr.Reference.Name)
+
+				// Assume the only reason for an error is because the RC is
+				// gone/missing, not for any other cause.  TODO(mml): something more
+				// sophisticated than this
+				if err == nil && ds != nil {
+					replicated = true
+				}
+			}
+		}
+		if replicated || o.Force {
+			pods = append(pods, pod)
+		}
+		if !replicated {
+			unreplicatedPodNames = append(unreplicatedPodNames, pod.Name)
+		}
+	}
+
+	if len(unreplicatedPodNames) > 0 {
+		joined := strings.Join(unreplicatedPodNames, ", ")
+		if !o.Force {
+			return pods, fmt.Errorf("Refusing to continue due to pods managed by neither a ReplicationController nor a DaemonSet: %s (use --force to override)", joined)
+		}
+		fmt.Fprintf(o.out, "WARNING: About to delete these pods not managed by a ReplicationController: %s\n", joined)
+	}
+	return pods, nil
+}
+
+// deletePods deletes the pods on the api server
+func (o *DrainOptions) deletePods(pods []api.Pod) error {
+	deleteOptions := api.DeleteOptions{}
+	if o.GracePeriodSeconds >= 0 {
+		gracePeriodSeconds := int64(o.GracePeriodSeconds)
+		deleteOptions.GracePeriodSeconds = &gracePeriodSeconds
+	}
+
+	for _, pod := range pods {
+		err := o.client.Pods(pod.Namespace).Delete(pod.Name, &deleteOptions)
+		if err != nil {
+			return err
+		}
+		cmdutil.PrintSuccess(o.mapper, false, o.out, "pod", pod.Name, "deleted")
+	}
+
+	return nil
+}
+
+// RunCordon runs either Cordon or Uncordon.  The desired value for "Unschedulable" is passed as the first arg.
+func (o *DrainOptions) RunCordon(desired bool) error {
```

Comment:
rename to RunCordonOrUncordon() to make it clearer. (also update comment with new name)